core/src/main/java/org/dozer/builder/xml/BeanMappingXMLBuilder.java

```diff
@@ -0,0 +1,143 @@
+
+    private final List<MappingsDefinition> mappingsDefinitions = new ArrayList<>();
+    private final JAXBModelParser<MappingsDefinition> factory;
+    private final ELEngine elEngine;
+    private final BeanContainer beanContainer;
+
+    public BeanMappingXMLBuilder(BeanContainer beanContainer, ELEngine elEngine) {
+        this.factory = new DefaultJAXBModelParser<>(beanContainer);
+        this.elEngine = elEngine;
+        this.beanContainer = beanContainer;
+    }
+
+    /**
+     * Loads the collection of files into memory and parses the XML
+     *
+     * @param files collection of file paths
+     */
+    public void loadFiles(List<String> files) {
+        for (String path : files) {
+            URL url = MappingValidator.validateURL(path, beanContainer);
+
+            LOG.info("Using URL [" + path + "] to load custom xml mappings");
+
+            try (InputStream stream = new BufferedInputStream(url.openStream())) {
+                load(stream);
+            } catch (IOException ex) {
+                MappingUtils.throwMappingException(ex);
+            }
+        }
+    }
+
+    /**
+     * Loads the collection of streams into memory and parses the XML
+     *
+     * @param xmlMappingSuppliers collection of streams to xml file/content
+     */
+    public void loadInputStreams(List<Supplier<InputStream>> xmlMappingSuppliers) {
+        for (Supplier<InputStream> supplier : xmlMappingSuppliers) {
+            try (InputStream stream = new BufferedInputStream(supplier.get())) {
+                load(stream);
+            } catch (IOException ex) {
+                MappingUtils.throwMappingException(ex);
+            }
+        }
+    }
+
+    /**
+     * Load the XML file from the stream
+     *
+     * @param stream stream to xml file/content - caller must close
+     */
+    private void load(InputStream stream) {
+        try {
+            String xmlContent = IOUtils.toString(stream, Charset.forName("UTF-8"));
+
+            MappingsDefinition result = factory.readXML(xmlContent, MappingsDefinition.class);
+            if (result == null) {
+                LOG.error("Failed to load custom xml mappings for: {}", xmlContent);
+            } else {
+                factory.validateXML(xmlContent);
+
+                mappingsDefinitions.add(new ELMappingsDefinition(elEngine, result));
+
+                LOG.info("Successfully loaded custom xml mapping.");
+            }
+        } catch (IOException ex) {
+            MappingUtils.throwMappingException(ex);
+        } catch (SAXException ex) {
+            MappingUtils.throwMappingException(ex);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<MappingFileData> build(BeanContainer beanContainer, DestBeanCreator destBeanCreator, PropertyDescriptorFactory propertyDescriptorFactory) {
+        List<MappingFileData> answer = new ArrayList<MappingFileData>();
```

Comment:
We could cleanup a bit here and use  the diamond operator.

Reply:
done.